src/main.cpp

```diff
@@ -4744,6 +4724,30 @@ PeerLogicValidation::PeerLogicValidation(CConnman* connmanIn) : connman(connmanI
     recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
 }
 
+void PeerLogicValidation::SyncTransaction(const CTransaction& tx, const CBlockIndex* pindex, int nPosInBlock) {
+    LOCK(cs_main);
```

Comment:
Yep, makes sense. But for now, it can at least be a short-circuit for returning without locking cs_main (not sure if there's any unlocked caller who would benefit, though).